ksql-version-metrics-client/src/test/java/io/confluent/ksql/version/metrics/KsqlVersionCheckerAgentTest.java

```diff
@@ -0,0 +1,195 @@
+  private ArgumentCaptor<BaseSupportConfig> configCaptor;
+  @Captor
+  private ArgumentCaptor<Supplier<Boolean>> activenessCaptor;
+
+  private KsqlVersionCheckerAgent ksqlVersionCheckerAgent;
+
+  @Before
+  public void setup() {
+    properties = new Properties();
+    properties.put("foo", "bar");
+    when(versionCheckerFactory.create(any(), any(), anyBoolean(), any())).thenReturn(ksqlVersionChecker);
+    ksqlVersionCheckerAgent = new KsqlVersionCheckerAgent(
+        activeQuerySupplier,
+        true,
+        clock,
+        versionCheckerFactory);
+  }
+
+  @Test
+  public void shouldStartTheAgentCorrectly() {
+
+    // When:
+    ksqlVersionCheckerAgent.start(KsqlModuleType.SERVER, properties);
+
+    // Then:
+    final InOrder inOrder = Mockito.inOrder(ksqlVersionChecker);
+    inOrder.verify(ksqlVersionChecker).init();
+    inOrder.verify(ksqlVersionChecker).setUncaughtExceptionHandler(any());
+    inOrder.verify(ksqlVersionChecker).start();
+  }
+
+  @Test (expected = Exception.class)
+  public void shouldFailIfVersionCheckerFails() {
+    // Given:
+    doThrow(new Exception("FOO")).when(ksqlVersionChecker).start();
+
+    // When:
+    ksqlVersionCheckerAgent.start(KsqlModuleType.SERVER, properties);
+  }
+
+  @Test
+  public void shouldCreateKsqlVersionCheckerWithCorrectConfigArg() {
+    // When:
+    ksqlVersionCheckerAgent.start(KsqlModuleType.SERVER, properties);
+
+    // Then:
+    verify(versionCheckerFactory).create(configCaptor.capture(), any(), anyBoolean(), any());
+    assertThat(configCaptor.getValue().getProperties().getProperty("foo"), equalTo("bar"));
+  }
+
+  @Test
+  public void shouldCreateKsqlVersionCheckerWithCorrectKsqlModuleType() {
+
+    // When:
+    ksqlVersionCheckerAgent.start(KsqlModuleType.SERVER, properties);
+
+    // Then:
+    verify(versionCheckerFactory).create(any(), eq(KsqlModuleType.SERVER), anyBoolean(), any());
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldCreateKsqlVersionCheckerWithCorrectActivenessStatusSupplier() {
+    // When:
+    ksqlVersionCheckerAgent.start(KsqlModuleType.SERVER, properties);
+
+    // Then:
+    verify(versionCheckerFactory).create(any(), any(), anyBoolean(), activenessCaptor.capture());
+    assertThat(activenessCaptor.getValue().get(), equalTo(true));
+  }
+
+  @Test
+  public void shouldReportActiveIfThereAreActiveQueries() {
+    // Given:
+    final Supplier<Boolean> activenessCheck = getActivenessCheck();
+
+    when(activeQuerySupplier.get()).thenReturn(true);
+
+    // Then:
+    assertThat(activenessCheck.get(), is(true));
+  }
+
+  @Test
+  public void shouldReportActiveIfThereAreRecentRequests() {
+    // Given:
+    final Supplier<Boolean> activenessCheck = getActivenessCheck();
+
+    givenLastRequest(NOW - ONE_DAY + 1); // <-- just recent enough to count.
+
+    // Then:
+    assertThat(activenessCheck.get(), is(true));
+  }
+
+  @Test
+  public void shouldReportInActiveIfThereAreNoActiveQueriesAndNoRequests() {
```

Comment:
```suggestion
  public void shouldReportInactiveIfThereAreNoActiveQueriesAndNoRequests() {
```
and similarly below.